File: bike-shop/bike-shop/backend/src/main/java/com/example/bike_shop/service/ContactService.java
package com.example.bike_shop.service;

import com.example.bike_shop.dto.ContactDTO;

import java.util.List;

public interface ContactService {
    ContactDTO submit(ContactDTO dto);
    List<ContactDTO> getAll();
    ContactDTO markProcessed(Long id);
    void delete(Long id);
}

File: bike-shop/frontend/src/services/contactService.js
import axiosClient from "./axiosConfig";

const contactService = {
  submit: (data) => axiosClient.post("/contact", data),
  getAll: () => axiosClient.get("/contact"),
  markProcessed: (id) => axiosClient.put(`/contact/${id}`),
  delete: (id) => axiosClient.delete(`/contact/${id}`),
};

export default contactService;
